[quant][graphmode][fx] Keep linear op unchanged when qconfig is not supported

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #48107 [quant][refactor] Move some util functions from torch/quantization/fx/utils.py to torch/quantization/utils.py
* #48073 [quant][fix][ez] Fix quant_type classification for fp16, fp16
* #48069 [quant][fix] Fix quant type classification for float_qparam qconfig
* **#48067 [quant][graphmode][fx] Keep linear op unchanged when qconfig is not supported**

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25008463](https://our.internmc.facebook.com/intern/diff/D25008463)

jerryzh168 committed:
[quant][graphmode][fx] Keep linear op unchanged when qconfig is not supported

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Keep linear op unchanged when qconfig is not supported"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25008463](https://our.internmc.facebook.com/intern/diff/D25008463)

[ghstack-poisoned]

supriyar commented on torch/quantization/fx/quantization_patterns.py:
```diff
@@ -301,8 +301,11 @@ def convert(self, quantizer, node, load_arg, debug=False, convert_custom_config_
-            " {}, supported dtypes are: {}".format(dtypes, supported_dtypes)
+        if dtypes not in supported_dtypes:
+            warnings.warn("dtype combination: {} is not "
+                          "supported by Linear".format(dtypes))
```
same suggestion regarding mentioning supported types.

jerryzh168 committed:
Update on "[quant][graphmode][fx] Keep linear op unchanged when qconfig is not supported"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25008463](https://our.internmc.facebook.com/intern/diff/D25008463)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Keep linear op unchanged when qconfig is not supported"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25008463](https://our.internmc.facebook.com/intern/diff/D25008463)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fx] Keep linear op unchanged when qconfig is not supported"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25008463](https://our.internmc.facebook.com/intern/diff/D25008463)

[ghstack-poisoned]
